compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -2237,6 +2240,38 @@ public BLangNode transform(RemoteMethodCallActionNode remoteMethodCallActionNode
         return bLangActionInvocation;
     }
 
+    @Override
+    public BLangNode transform(StreamTypeDescriptorNode streamTypeDescriptorNode) {
+        String streamTypeName = "stream";
+        String anyTypeName = "any";
+        BLangType constraint, error = null;
+        Optional<Node> paramsNode = streamTypeDescriptorNode.streamTypeParamsNode();
+        boolean hasConstraint = paramsNode.isPresent();
+        boolean hasError = hasConstraint && ((StreamTypeParamsNode) paramsNode.get()).rightTypeDescNode().isPresent();
+        DiagnosticPos pos = getPosition(streamTypeDescriptorNode);
+
+        if (hasError) {
+            error = (BLangType) ((StreamTypeParamsNode) paramsNode.get()).rightTypeDescNode().get().apply(this);
+        }
+        if (!hasConstraint) {
+            constraint = addValueType(pos, anyTypeName);
+        } else {
+            constraint = (BLangType) ((StreamTypeParamsNode) paramsNode.get()).leftTypeDescNode().apply(this);
```

Comment:
`((StreamTypeParamsNode) paramsNode.get())` is called several times. consider extracting it to a variable.